### DISCUSSION TITLE
zfs property with no value.

### DISCUSSION THREAD

**Discussion by gpetrom** [psy0rz/zfs_autobackup#241] [Ideas]
Hi 

I am testing zfs-autobackup and i was trying to see how it works in one dataset.
I have a pool storage with some datasets inside for example 100,101,102,103
So i configured the property  with the following command
zfs set autobackup:offsite1=true storage/103 
When i runned zfs-autobackup it started to delete snapshots on target on the other datasets.
Is this normal behavior? 
Maybe zfs-autobackup would ignore datasets that has the property but they have no value inside

Thanks in advanced

**Comment by psy0rz** [comment]
Please shiw the commandline youre using and the verbose output. (Use --test to prevent it doing anything)

**Reply by gpetrom** [reply]
Thanks for the reply 
I have made a test env to test it.
So i have enable daily only in storage 104

Property values
`
(backup) root@debian12:~# zfs get -t filesystem,volume autobackup:daily
NAME         PROPERTY                 VALUE                    SOURCE
storage      autobackup:daily  -                 -
storage/100  autobackup:daily  -                 -
storage/101  autobackup:daily  -                 -
storage/102  autobackup:daily  -                 -
storage/103  autobackup:daily  -                 -
storage/104  autobackup:daily  parent            local
storage/105  autobackup:daily  -  
`
zfs-autobackup -v --clear-mountpoint --ssh-target debian11_test hourly storage/images --test --no-holds --strip-path=1 --keep-source=2 --keep-target=2 --min-change=0 --snapshot-format=autosnap_%Y-%m-%d_%H:%M:%S_daily --buffer 128M --utc
`
With --test 
`
(backup) root@debian12:~# zfs-autobackup -v --clear-mountpoint --ssh-target debian11_test hourly storage/images --test --no-holds --strip-path=1 --keep-source=2 --keep-target=2 --min-change=0 --snapshot-format=autosnap_%Y-%m-%d_%H:%M:%S_daily --buffer 128M --utc
  zfs-autobackup v3.2 - (c)2022 E.H.Eefting (edwin@datux.nl)
  
  NOTE: TEST MODE - SIMULATING WITHOUT MAKING ANY CHANGES
  
  Selecting dataset property : autobackup:hourly
  Snapshot format            : autosnap_%Y-%m-%d_%H:%M:%S_daily
  Timezone                   : UTC
  
  #### Source settings
  [Source] Keep the last 2 snapshots.
  
  #### Selecting
  [Source] storage/102: Selected
  [Source] storage/103: Selected
  [Source] storage/104: Selected
  
  #### Snapshotting
  [Source] Creating snapshots autosnap_2024-01-15_10:22:31_daily in pool storage
  
  #### Target settings
  [Target] SSH to: debian11_test
  [Target] Keep the last 2 snapshots.
  [Target] Receive datasets under: storage/images
  
  #### Synchronising
  [Source] zfs send buffer        : 128M
  [Target] zfs recv buffer        : 128M
  [Source] storage/102: sending to storage/images/102
  [Source] storage/102@autosnap_2024-01-14_11:08:03_daily: Destroying
  [Source] storage/102@autosnap_2024-01-14_11:15:37_daily: Destroying
  [Target] storage/images/102@autosnap_2024-01-14_11:08:03_daily: Destroying
  [Target] storage/images/102@autosnap_2024-01-14_11:15:37_daily: Destroying
  [Target] storage/images/102@autosnap_2024-01-15_10:22:31_daily: receiving incremental
  [Source] storage/103: sending to storage/images/103
  [Source] storage/103@autosnap_2024-01-14_11:08:03_daily: Destroying
  [Source] storage/103@autosnap_2024-01-14_11:15:37_daily: Destroying
  [Target] storage/images/103@autosnap_2024-01-14_11:08:03_daily: Destroying
  [Target] storage/images/103@autosnap_2024-01-14_11:15:37_daily: Destroying
  [Target] storage/images/103@autosnap_2024-01-15_10:22:31_daily: receiving incremental
  [Source] storage/104: sending to storage/images/104
  [Source] storage/104@autosnap_2024-01-14_11:08:03_daily: Destroying
  [Source] storage/104@autosnap_2024-01-14_11:15:37_daily: Destroying
  [Target] storage/images/104@autosnap_2024-01-14_11:08:03_daily: Destroying
  [Target] storage/images/104@autosnap_2024-01-14_11:15:37_daily: Destroying
  [Target] storage/images/104@autosnap_2024-01-15_10:22:31_daily: receiving incremental
  > Analysing missing 2/2
  #### All tests successful.

`
Without --test 

`
(backup) root@debian12:~# zfs-autobackup -v --clear-mountpoint --ssh-target debian11_test hourly storage/images --no-holds --strip-path=1 --keep-source=2 --keep-target=2 --min-change=0 --snapshot-format=autosnap_%Y-%m-%d_%H:%M:%S_daily --buffer 128M --utc
  zfs-autobackup v3.2 - (c)2022 E.H.Eefting (edwin@datux.nl)
  
  
  Selecting dataset property : autobackup:hourly
  Snapshot format            : autosnap_%Y-%m-%d_%H:%M:%S_daily
  Timezone                   : UTC
  
  #### Source settings
  [Source] Keep the last 2 snapshots.
  
  #### Selecting
  [Source] storage/102: Selected
  [Source] storage/103: Selected
  [Source] storage/104: Selected
  
  #### Snapshotting
  [Source] Creating snapshots autosnap_2024-01-15_10:24:34_daily in pool storage
  
  #### Target settings
  [Target] SSH to: debian11_test
  [Target] Keep the last 2 snapshots.
  [Target] Receive datasets under: storage/images
  
  #### Synchronising
  [Source] zfs send buffer        : 128M
  [Target] zfs recv buffer        : 128M
  [Source] storage/102: sending to storage/images/102
  [Source] storage/102@autosnap_2024-01-14_11:08:03_daily: Destroying
  [Source] storage/102@autosnap_2024-01-14_11:15:37_daily: Destroying
  [Target] storage/images/102@autosnap_2024-01-14_11:08:03_daily: Destroying
  [Target] storage/images/102@autosnap_2024-01-14_11:15:37_daily: Destroying
  [Target] storage/images/102@autosnap_2024-01-15_10:24:34_daily: receiving incremental
  [Source] storage/103: sending to storage/images/103
  [Source] storage/103@autosnap_2024-01-14_11:08:03_daily: Destroying
  [Source] storage/103@autosnap_2024-01-14_11:15:37_daily: Destroying
  [Target] storage/images/103@autosnap_2024-01-14_11:08:03_daily: Destroying
  [Target] storage/images/103@autosnap_2024-01-14_11:15:37_daily: Destroying
  [Target] storage/images/103@autosnap_2024-01-15_10:24:34_daily: receiving incremental
  [Source] storage/104: sending to storage/images/104
  [Source] storage/104@autosnap_2024-01-14_11:08:03_daily: Destroying
  [Source] storage/104@autosnap_2024-01-14_11:15:37_daily: Destroying
  [Target] storage/images/104@autosnap_2024-01-14_11:08:03_daily: Destroying
  [Target] storage/images/104@autosnap_2024-01-14_11:15:37_daily: Destroying
  [Target] storage/images/104@autosnap_2024-01-15_10:24:34_daily: receiving incremental
  > Analysing missing 2/2
  #### All operations completed successfully

`

**Comment by psy0rz** [comment]
Yes this is exactly the intended behaviour. 

It will thin out all the snapshots in the target path that match the snapshot-format. It doesnt look at properties.

**Reply by gpetrom** [reply]
Maybe you consider fixing this, a lot of errors could happen and if the target is local that is ok but if you push it in an offsite server that could cause problems.
One more example that could cause a lot of problems.
I have two servers proxmox A and proxmox B 
i have in proxmox A vms 100,101,102 running in storage/proxmoxA   
i have in proxmox B vms 200,201,202 running in  storage/proxmoxB  
and i want to backup Proxmox A to Proxmox B and vice versa.
If i forgot to make In Proxmox A the storage storage/proxmoxB false and just enable the backup storage/proxmoxA. 
zfs-autobackup will delete source snapshots in  Proxmox B.

I know that this is extreme scenario but it could happen

**Comment by psy0rz** [comment]
there is nothing to fix, since it aint broken :)

just use the correct thinning schema so it wont thin stuff you want to keep.

if you need two different thinning schemas, either use a different target path for each server, or use a different backup name.
